pybamm/parameters/parameter_values.py

```diff
@@ -638,28 +662,60 @@ def _process_symbol(self, symbol):
+                    name, data = function_name
+                    function = pybamm.Interpolant(
+                        data[:, 0], data[:, 1], *new_children, name=name
                     )
-                )
-                self.parameter_events.append(
-                    pybamm.Event(
-                        "Interpolant {} upper bound".format(name),
-                        pybamm.min(max(data[:, 0]) - new_children[0]),
-                        pybamm.EventType.INTERPOLANT_EXTRAPOLATION,
+                    # Define event to catch extrapolation. In these events the sign is
+                    # important: it should be positive inside of the range and negative
+                    # outside of it
+                    self.parameter_events.append(
+                        pybamm.Event(
+                            "Interpolant {} lower bound".format(name),
+                            pybamm.min(new_children[0] - min(data[:, 0])),
+                            pybamm.EventType.INTERPOLANT_EXTRAPOLATION,
+                        )
                     )
-                )
+                    self.parameter_events.append(
+                        pybamm.Event(
+                            "Interpolant {} upper bound".format(name),
+                            pybamm.min(max(data[:, 0]) - new_children[0]),
+                            pybamm.EventType.INTERPOLANT_EXTRAPOLATION,
+                        )
+                    )
+
+                elif len(function_name) == 3:  # ML data
+                    # to create an Interpolant
+                    name, data, _ = function_name
+
+                    function = pybamm.Interpolant(
+                        data[0], data[-1], new_children, name=name
+                    )
+                    # Define event to catch extrapolation. In these events the sign is
+                    # important: it should be positive inside of the range and negative
+                    # outside of it
+                    for data_index in range(len(data[0]) - 1):
+                        self.parameter_events.append(
+                            pybamm.Event(
+                                "Interpolant {} lower bound".format(name),
+                                pybamm.min(new_children[data_index] - min(data[0][data_index])),
+                                pybamm.EventType.INTERPOLANT_EXTRAPOLATION,
+                            )
+                        )
+                        self.parameter_events.append(
+                            pybamm.Event(
+                                "Interpolant {} upper bound".format(name),
+                                pybamm.min(max(data[0][data_index]) - new_children[data_index]),
+                                pybamm.EventType.INTERPOLANT_EXTRAPOLATION,
+                            )
+                        )
+
+                else:
+                    raise Exception("Invalid function name length: {0}".format(len(function_name)))
```

Comment:
I don't think this exception is needed, since this is an internal (not user-facing) function.

Reply:
It is an internal exception, but I like to use such exceptions. If I have multiple `if` - `elif` - `elif` statements I always consider the possibility that there might somewhere be a problem or bug, and without the `else` statement we have undefined behavior. Here it is not so critical but especially in Web apps it is important, not doing it can lead to vulnerabilities like DoS

Reply:
If you can write a test that triggers this line of code then we can keep it. Also, please use a more specific exception